trunner/config.py

```diff
@@ -30,73 +37,135 @@ def __init__(self, parser, message="Parser error"):
+
+    def parse_harness(self, test):
+        test['type'] = 'harness'
+        test['harness'] = self.path / test['harness']
+
+        if not test['harness'].exists():
+            raise YAMLParserError(self, f"harness {test['harness']} file not found")
+
+        if not str(test['harness']).endswith('.py'):
+            raise YAMLParserError(self, "harness file must has .py extension")
 
+    @staticmethod
+    def parse_array_value(array):
+        values = set(array.get('value', []))
+        values |= set(array.get('include', []))
+        values -= set(array.get('exclude', []))
+
+        return values
+
+    def parse_target(self, test):
+        target_array = test.get('target', dict())
+        target_array.setdefault('value', DEFAULT_TARGETS)
+        targets = YAMLParser.parse_array_value(target_array)
+
+        not_known = targets - set(ALL_TARGETS)
+        if not_known:
+            raise YAMLParserError(self, f"wrong target: {', '.join(map(str, not_known))}")
+
+        targets &= set(self.targets)
+        test['target'] = list(targets)
+
+    def parse_test_case(self, test):
+        self.parse_keywords(test)
+        self.parse_name(test)
+        self.parse_target(test)
         # If type is not known assume that it's a unit test
-        config.setdefault('type', 'unit')
-        config.setdefault('ignore', False)
-        config.setdefault('timeout', PYEXPECT_TIMEOUT)
+        test.setdefault('type', 'unit')
+        test.setdefault('timeout', PYEXPECT_TIMEOUT)
 
-        if not isinstance(config['ignore'], bool):
-            config['ignore'] = False
+        test.setdefault('ignore', False)
+        if not isinstance(test['ignore'], bool):
+            test['ignore'] = False
 
-        if config.get('harness'):
-            config['type'] = 'harness'
-            config['harness'] = self.path / config['harness']
+        if test.get('harness'):
+            self.parse_harness(test)
 
-            if not config['harness'].exists():
-                raise YAMLParserError(self, f"harness {config['harness']} file not found")
+        # Copy test per target
+        test_case = []
+        for target in test['target']:
+            test_by_target = test.copy()
+            test_by_target['target'] = target
+            test_case.append(test_by_target)
 
-            if not str(config['harness']).endswith('.py'):
-                raise YAMLParserError(self, "harness file must has .py extension")
+        return test_case
 
-        return config
+    def inherit_array_value(self, test, keyword):
+        if keyword not in test or keyword not in self.config:
+            return
+
+        array_keywords = ('value', 'include', 'exclude')
+        global_array = self.config[keyword]
+        array = test[keyword]
+
+        for key in array_keywords:
+            inherited = global_array.get(key, [])
+            if inherited:
+                array.setdefault(key, []).extend(inherited)
```

Comment:
I would go for computing global array and setting it for test->array->values 

For me intuitive would be to:

```yaml
test:
  target:
    value: [a, b]
    include: [c]
  tests:
    - name: test1  # expected to be run on d
      target:
        value: [d]
    - name: test2  # expected to be run on a, b, c, e
      target:
        include: [e]
    - name: test3  # expected to be run on a, b
       target:
        exclude: [c]

```
If I read correctly the current code:
```yaml
test1 # run on a, b, c, d
test2 # run on a, b, c, e
test3 # run on a, b (but only because of implementation detail)
```

FYI, first I've used `targets` keyword instead of `target` as it was better mnemonic for "array of targets". Fixed this only after re-reading the source code. Maybe `targets` would be better?

Reply:
Now targets should inherit in more intuitive way. I'm a little bit dissatisfied with `YAMLParser` as a whole -- is hard to test it in more "modular" way, but it should be a topic for another PR.